src/Database/Esqueleto/Internal/Internal.hs

```diff
@@ -1835,14 +1833,15 @@ data SideData = SideData
     , sdLimitClause    :: !LimitClause
     , sdLockingClause  :: !LockingClause
     , sdCteClause      :: ![CommonTableExpressionClause]
+    , sdReturningClause :: !ReturningClause
```

Comment:
I wonder if `Maybe ReturningClause` is a better type here. Given a satisfactory `Semigroup ReturningClause`, we get the desired `Monoid` for free from `Maybe`.